### DISCUSSION TITLE
python3 with USB breakout boards like the MCP2221 or FT232H?

### DISCUSSION THREAD

**Discussion by freemansoft** [micropython/micropython#10504] [Unix & Windows]
What is the analog to CircuitPython's Blinka for Windows machines? Is there a way of using Micropython I2C/GPIO/Analog/SPI libraries on a PC running CPython or MicroPython.

I'd like to build Jupyter notebooks that talk with peripherals.  I think that there would be 3 options.

1.  Use CPython with Blinka shim and a breakout board.  
    * Can run standard Jupyter notebook with CPython.
3.  Use MicroPython with some kind of adapter to the breakout board.  
    * Can't find any reference to this.
5.  Run Jupyter Notebook kernel that essentially runs the REPL to a remote board. In my case, I'd probably use an RP2040. basically as a peripheral to the Jupyter notebook. This has a complication in that you have to manually manage getting files/libraries onto the remote device so it isn't really _transparent_ that you are working on the microcontroller.
    1. https://github.com/Carglglz/jupyter_upydevice_kernel
    2. https://github.com/goatchurchprime/jupyter_micropython_kernel/

I'm currently running native CPython on a windows machine talking with I2C peripherals that are attached to an MCP2221 breakout board that connects to my PC over USB.  This lets me interact with I2C devices using any of the standard Python toolings on my PC. My setup is here https://joe.blog.freemansoft.com/2023/01/using-circuitpython-and-adafruit.html.

**Comment by cve2022** [comment]
Have a look at [belay](https://github.com/BrianPugh/belay).

**Reply by freemansoft** [reply]
Thanks.  Will do.

**Reply by freemansoft** [reply]
Belay is interesting. You tag code to be pushed to the microcontroller.  The PC has RPC-style endpoints on your PC that act as proxies for the functions on the controller.  Then you mix in code on your PC that interacts with the remote functions via the endpoints.  You have to push your libraries to the microcontrollers because the `@Task`  code actually runs there.  The example files are a mix of PC and Microcontroller code and functions. I'm not clear on how Python classes are tagged and managed. 

I'm not sure I'm smart enough to use it.
